File: src/pokemon.cpp
//
// Created by kubun on 06/03/23.
//
#include <iostream>
#include <utility>
#include "pokemon.h"

Pokemon::Pokemon(std::string name, int atk, int Lev, int Precision) : pokename(std::move(name)), Attack(atk), Level(Lev), Accurracy(Precision){};

std::string Pokemon::get_pokename() const {
    return Pokemon::pokename;
}
std::string Pokemon::set_pokename() {
    std::string newname;
    std::cin>>newname;
    Pokemon::get_pokename();
    pokename = newname;
    return pokename;
}
void Pokemon::show_data() const {
    std::cout<<"Level is "<<Level;
}

Pokemon::Pokemon(const Pokemon &pokemon) {

}

Pokemon::~Pokemon() = default;

int main(){
    Pokemon Dracaufeu {"Dracaufeu", 100, 1, 100};
    Dracaufeu.set_pokename();
    Dracaufeu.show_data();
    return 0;
}

File: python/pokemon.py
import random
import sys


class Pokemon:
    def __init__(self, name, atk, level, accuracy):
        self.__name = name
        self.__hp = 100
        self.attack = atk
        self.defence = 0
        self.level = level
        self.accuracy = accuracy

    def get_name(self):
        return self.__name

    def get_hp(self):
        return self.__hp

    def set_name(self):
        self.get_name()
        newname = input("Entrer le nouveau nom du pokémon.")
        self.__name = newname

    def set_hp(self):
        self.get_hp()


